# HG changeset patch
# User Daniel Serpell <daniel.serpell@gmail.com>
# Date 1258942903 10800
# Branch trunk
# Node ID 6b4836687972bd6c038aa0f1caa3c43172be973f
# Parent  768944c3d33626e5274e42794b1a078b5fdea258
Simplify based on cbm routine. 3 bytes less.

diff --git a/libsrc/atari/open.s b/libsrc/atari/open.s
--- a/libsrc/atari/open.s
+++ b/libsrc/atari/open.s
@@ -27,12 +27,11 @@
 
 .proc	_open
 
-	cpy	#4		; correct # of arguments (bytes)?
+	dey			; parm count < 4 shouldn't be needed to be checked
+	dey			;	(it generates a c compiler warning)
+	dey
+	dey
 	beq	parmok		; parameter count ok
-	tya			; parm count < 4 shouldn't be needed to be checked
-	sec			;	(it generates a c compiler warning)
-	sbc	#4
-	tay
 	jsr	addysp		; fix stack, throw away unused parameters
 
 parmok:	jsr	findfreeiocb
